### PR TITLE
fix: storybook rendering issue

### PR DESCRIPTION
**Pull request by keita-determined**
## Description

[DET-7759]

## Test Plan

1. Run storybook to see if it works. I tested with `npm run storybook`
2. Check the theme (Light mode or Dark mode)

## Commentary (optional)

~~On my environment, theme is always dark mode. Not sure if someone else gets the same issue or not. 
The background color is white, so I cannot see white text in dark mode.~~ 
Theme on Storybook depends on system mode

Also, I added `StoreProvider` and `BrowserRouter` in the top level because I saw some errors that come from them. Instead of adding these in each story, I thought adding these in the top level is a good idea to avoid the errors in the future. For example, `context`  related hooks would be used in some components but forget to add `decorators` in the story etc...

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-7759]: https://determinedai.atlassian.net/browse/DET-7759?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/Dropdown.stories.tsx [hunk 1]
```diff
@@ -25,83 +25,85 @@ export const Default = (): React.ReactNode => (
...
-          </Dropdown>
-        </td>
-        <td align="right">
-          <Dropdown content={content} placement={Placement.BottomRight}>
-            <Button>Bottom Right</Button>
-          </Dropdown>
-        </td>
-        <td />
-      </tr>
+      <tbody>
```
Comment (keita-determined) [inline webui/react/src/components/Dropdown.stories.tsx:28]: Added `tbody` due to warning in console

**Comment by keita-determined** [comment]
`build-storybook` failed due to `Error 137` (Not enough memory).

**Comment by hamidzr** [comment]
we should figure out why that's happening, is it temproray or happening consistently? maybe we need to bump the ci resources or tweak the build config (if available)

**Comment by keita-determined** [comment]
> thanks for addressing the issue. we should figure out why the ci build failure is happening, is it temproray or happening consistently? maybe we need to bump the ci resources or tweak the build config (if available)

I ran the ci job several time and failed all, so it's happening consistently.

**Comment by hamidzr** [comment]
checks out  https://app.circleci.com/pipelines/github/determined-ai/determined/24448/workflows/de8e9a01-4e08-4b5c-a954-da1489decf5d/jobs/791348/resources
https://circleci.com/docs/using-docker#available-docker-resource-classes 

we could try bumping the resources. or look to 

or if we wanna dig further we can see
- what was the mem usage prior to this pr
- is the package upgrade the cause?
- see if storybook gives us options to control this. 
- time the build time with different resource classes

ci config file is here /.circleci/config.yaml

**Comment by keita-determined** [comment]
Compared to previous CircleCI tasks, CPU usage, memory usage and running time are pretty much similar
- [This PR](https://app.circleci.com/pipelines/github/determined-ai/determined/24448/workflows/de8e9a01-4e08-4b5c-a954-da1489decf5d/jobs/791348/resources)
- [PR 4555](https://app.circleci.com/pipelines/github/determined-ai/determined/24467/workflows/bb55f6f9-b76b-46d9-839d-f89b8888b2df/jobs/791920/resources)
- [PR 4551](https://app.circleci.com/pipelines/github/determined-ai/determined/24421/workflows/6c220bba-3272-4351-a893-3583b8f63f99/jobs/790214/resources)

## Plan
- Find options to handle this on Storybook
- Revert npm libarry update commit
- Upgrade the resource on CircleCI

**Comment by hamidzr** [comment]
> Upgrading the resource from medium to medium+ works. I think this approach is better than reverting library. 

sounds good to me thanks for investigating it. i'm curious to see if this also sped up the job

**Review by hamidzr — COMMENTED** [review]
the changes look good to me overall thanks for addressing the issues. It seems there is a small themeing issue left. some text disappear after a few seconds when I open storybooks and it looks like to be because the font color changes and the background doesn't.

https://output.circle-artifacts.com/output/job/e02f662c-8144-4a16-9494-988f8b720dca/artifacts/0/webui/react/build-storybook/index.html?path=/story/humanreadablenumber--positive-infinity

**Comment by keita-determined** [comment]
> > Upgrading the resource from medium to medium+ works. I think this approach is better than reverting library.
> 
> sounds good to me thanks for investigating it. i'm curious to see if this also sped up the job

`build-storybook` task doesn't take that long (from 1.5min  to 3min). https://app.circleci.com/pipelines/github/determined-ai/determined/24492/workflows/ac57c9e1-0721-4b3e-9eee-19ba08f3ec0d/jobs/792910

[test-det-deploy-local](https://circleci.com/gh/determined-ai/determined/792976?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link) takes long time

**Comment by hamidzr** [comment]
1':17" to 58s

**Comment by keita-determined** [comment]
> the changes look good to me overall thanks for addressing the issues. It seems there is a small themeing issue left. some text disappear after a few seconds when I open storybooks and it looks like to be because the font color changes and the background doesn't.
> 
> https://output.circle-artifacts.com/output/job/e02f662c-8144-4a16-9494-988f8b720dca/artifacts/0/webui/react/build-storybook/index.html?path=/story/humanreadablenumber--positive-infinity

Are you using dark mode on Mac? If so, that's the reason. The theme depends on local mode.
Should I fix this issue this PR or make a ticket and new PR?

File: webui/react/.storybook/preview.js [hunk 2]
```diff
@@ -1,8 +1,27 @@
-import 'shared/styles/index.scss';
-import 'shared/styles/storybook.scss';
```
Comment (hamidzr) [inline webui/react/.storybook/preview.js:2]: ' or " ? let's use whichever we use elsewhere

**Comment by hamidzr** [comment]
ah, you're right. the light theme works better. there is a delay for the theme to get applied. I'm curious to see what would we need to address that

File: webui/react/.storybook/preview.js (hunk 2 above)
Comment (keita-determined) [reply webui/react/.storybook/preview.js:2]: We should define the rule in ESLint.

Comment (keita-determined) [reply webui/react/.storybook/preview.js:2]: Fixed!

**Comment by keita-determined** [comment]
Enabled dark mode anyways
image

File: webui/react/src/shared/styles/storybook.scss [hunk 3]
```diff
@@ -8,6 +8,7 @@
  */
 
 #root {
+  background-color: var(--theme-stage);
```
Comment (keita-determined) [inline webui/react/src/shared/styles/storybook.scss:11]: added this background color. Another option is `--theme-background` but it was too dark for dark mode, and this color isn't commonly visible in main screen of determined.
Any thoughts?

**Comment by keita-determined** [comment]
Fixed an issue in `LogViewer`.
It was just a wrong decoder.

For `analytics is not defined`, it comes from [here](https://github.com/determined-ai/determined/blob/master/webui/react/src/utils/error.ts#L83). 
Even though there is an error in StoryBook, you can see the stories when you click `close` button in the bottom

File: webui/react/src/components/LogViewer/LogViewer.stories.tsx [hunk 4]
```diff
@@ -23,7 +22,7 @@ const messageWithTags = `continuing trial: <COMPUTE_VALIDATION_METRICS: (10,10,3
 export const Default = (): React.ReactNode => (
   <div style={{ height: '100vh' }}>
     <LogViewer
-      decoder={jsonToClusterLog}
+      decoder={mapV1LogsResponse}
```
Comment (hamidzr) [inline webui/react/src/components/LogViewer/LogViewer.stories.tsx:25]: what motivated this?

File: webui/react/.storybook/preview.js [hunk 5]
```diff
@@ -2,7 +2,69 @@ import 'shared/styles/index.scss';
...
+import { BrowserRouter } from 'react-router-dom';
+import { useEffect } from 'react';
 
-export const decorators = [ ThemeDecorator ];
+export const globalTypes = {
+  // https://storybook.js.org/addons/@storybook/addon-toolbars
+  theme: {
+    name: 'Theme',
+    description: 'Global theme for components',
+    defaultValue: ThemeOptions.light.displayName,
```
Comment (hamidzr) [inline webui/react/.storybook/preview.js:16]: nit default to the system? if that's what we do in core. or maybe it's better to keep it at "light" always 🤔

**Review by hamidzr — COMMENTED** [review]
I don't see the delay that we saw in the previous iteration. LGTM 
https://output.circle-artifacts.com/output/job/667dfafa-8133-4fb7-b074-0fbecc4da063/artifacts/0/webui/react/build-storybook/index.html?path=/story/slotallocationbar--no-legends

**Review by hamidzr — APPROVED** [review]

File: webui/react/src/components/LogViewer/LogViewer.stories.tsx (hunk 4 above)
Comment (keita-determined) [reply webui/react/src/components/LogViewer/LogViewer.stories.tsx:25]: LogView doesn't display with `jsonToClusterLog`, so I changed it to `mapV1LogsResponse`

File: webui/react/.storybook/preview.js (hunk 5 above)
Comment (keita-determined) [reply webui/react/.storybook/preview.js:16]: I'm fine with either way. @hkang1 any preference?

**Comment by keita-determined** [comment]
> I don't see the delay that we saw in the previous iteration. LGTM
> https://output.circle-artifacts.com/output/job/667dfafa-8133-4fb7-b074-0fbecc4da063/artifacts/0/webui/react/build-storybook/index.html?path=/story/slotallocationbar--no-legends

Surprisingly, `analytics` error doesn't show up in here. It happens in local though.

**Pull request merged**